taxonomy/management/commands/refresh_xblock_skills.py

```diff
@@ -0,0 +1,124 @@
+        $ # args-from-database means command line arguments will be picked from the database.
+        $ ./manage.py refresh_xblock_skills --args-from-database
+        $ # To update all xblocks in all the courses
+        $ ./manage.py refresh_xblock_skills --all --commit
+    """
+    help = 'Refreshes the skills associated with XBlocks.'
+    product_type = ProductTypes.XBlock
+
+    def add_arguments(self, parser):
+        """
+        Add arguments to the command parser.
+        """
+        parser.add_argument(
+            '--course',
+            metavar=_('COURSE_KEY'),
+            action='append',
+            help=_('Update skills for XBlocks under given course keys.'),
+            default=[],
+        )
+        parser.add_argument(
+            '--xblock',
+            metavar=_('USAGE_KEY'),
+            action='append',
+            help=_('Update skills for given Xblock usage keys.'),
+            default=[],
+        )
+        parser.add_argument(
+            '--args-from-database',
+            action='store_true',
+            help=_('Use arguments from the RefreshXBlockSkillsConfig model instead of the command line.'),
+        )
+        parser.add_argument(
+            '--all',
+            action='store_true',
+            help=_('Create xblock skill mapping for all xblocks in all the courses.'),
+        )
+        parser.add_argument(
+            '--commit',
+            action='store_true',
+            default=False,
+            help=_('Commits the skills to storage.')
+        )
+
+    def get_args_from_database(self):
+        """
+        Return an options dictionary from the current RefreshXBlockSkillsConfig model.
+        """
+        config = RefreshXBlockSkillsConfig.get_solo()
+        argv = config.arguments.split()
+        parser = self.create_parser('manage.py', 'refresh_xblock_skills')
+        return parser.parse_args(argv).__dict__
+
+    def handle(self, *args, **options):
+        """
+        Entry point for management command execution.
+        """
+        if not (options['args_from_database'] or options['all'] or options['course'] or options['xblock']):
+            raise InvalidCommandOptionsError(
+                'Either course, xblock, args_from_database or all argument must be provided.',
+            )
+
+        if options['args_from_database']:
+            options = self.get_args_from_database()
+
+        if options['course'] and options['xblock']:
+            raise InvalidCommandOptionsError('Either course or xblock argument should be provided and not both.')
+
+        LOGGER.info('[TAXONOMY] Refresh XBlock Skills. Options: [%s]', options)
+
+        courses = []
+        xblocks_from_args = []
+        xblock_provider = get_xblock_metadata_provider()
+        if options['all']:
+            courses = get_course_metadata_provider().get_all_courses()
+        elif options['course']:
+            courses = [{"key": course} for course in options['course']]
```

Comment:
Are we supposed to pass a course key or a course id here? The usage examples have both. It should raise an exception if it's not in the required format. Would be helpful if we can add this info in the help text as well.

Reply:
@sameenfatima78 It requires a `course-key`.

>The usage examples have both.

Updated this:
`$ ./manage.py refresh_xblock_skills --course 'course-v1:edX+DemoX+Demo_Course' --course 'Course2_key' --commit`
to 
`$ ./manage.py refresh_xblock_skills --course 'course-v1:edX+DemoX+Demo_Course' --course 'course-v2:edX+DemoX+Demo_Course2' --commit`

>It should raise an exception if it's not in the required format.

Should we include `opaque_keys` as dependency and use `course_key = CourseKey.from_string(course_key_str)` as mentioned in 
https://github.com/openedx/edx-platform/wiki/Opaque-Keys-(Locators)#parse-course-ids

Reply:
I think it's better to add a check right in the beginning before any service to service calls are made. We can add Opaque Keys as dependency here.

Reply:
@sameenfatima78 Done. Also validated xblock usage_keys.